tests/models/test_trigger.py

```diff
@@ -101,6 +101,9 @@ def test_submit_event(session, create_task_instance):
     session.commit()
     # Call submit_event
     Trigger.submit_event(trigger.id, TriggerEvent(42), session=session)
+    # commit changes made by submit event and expire all cache to read from db.
+    session.commit()
+    session.expire_all()
```

Comment:
```suggestion
    session.flush()
```

Should be enough I think?

Reply:
We will atleast need the following, flush won't mark the state as out-of-date, so I think `session.expire_all` is fine tbh:

```
    session.flush()
    session.expire(task_instance)
```

Reply:
Better -- I have a pet peave of avoiding `commits()` where possible in tests as I have a dream of one day running most tests in an auto-rollingback transaction for speedier setup/teardown.